Week3/SOLUTION

```diff
@@ -0,0 +1,106 @@
+#1. Create table
+import sqlite3 as sq
+con=sq.connect("c:\\sqlite\\EMS.db")
+q="create table employee(e_id int primary key,e_name text,dept text,basic int ,branch text)"
+cur=con.cursor()
+cur.execute(q)
+
+#2. Insert 5 Records directly, 5 records using tuple and 5 records using taking input from user
+
+#insert record directy
+
+q1="insert into employee values(1,'jay','Account',5000,'surat'),(2,'suresh','IT',6000,'vyara'),(3,'rahul','HR',7000,'bardoli'),(4,'hiren','IT',8000,'surat'),(5,'dinesh','Inventory',3000,'bardoli')"
+cur=con.cursor()
+cur.execute(q)
+
+# insert record using tuple
+
+q="insert into employee values(?,?,?,?,?)"
+v=[(6,'rakesh','inventory',7000,'surat'),(7,'avinash','Account',5000,'bardoli'),(8,'arvind','HR',4000,'vyara'),(9,'vatsal','IT',9000,'baruch'),(10,'eren','Account',6000,'surat')]
+cur=con.cursor()
+cur.executemany(q,v)
+
+#insert record using user input
+q="insert into employee values(?,?,?,?,?)"
+l=[]
+for i in range(5):
+    eid=int(input("Enter employee id:"))
+    ename=input("Enter name:")
+    dept=input("Enter departnment:")
+    basic=int(input("Enter basic salary:"))
+    branch=input("Enter branch:")
+    t=(eid,ename,dept,basic,branch)
+    l.append(t)
```

Comment:
please explain me this whole for loop and why did you use append in last and what reason is it to use??

Reply:
The for loop iterates through each row retrieved from the database result set. The append() function is used to add each row to a list, data_list. This allows you to store and work with the retrieved data outside the loop, making it accessible for further processing or analysis, rather than just printing it to the console.